### PR TITLE
Pipeline: support `collectRequestUnit`

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #6518

Problem Summary:

### What is changed and how it works?
- add `QueryProfileInfo` to aggregate all `TaskProfileInfo`.
- Assuming that `QueryProfileInfo::cpu_execute_time_ns` is approximately equal to cputime, and using `QueryProfileInfo::cpu_execute_time_ns` to calculate RU.
- add setting `force_enable_pipeline`.

### Check List

Tests 

- [x] compute side tsan passed
  - https://ci.pingcap.net/blue/organizations/jenkins/tiflash-sanitizer-daily/detail/tiflash-sanitizer-daily/1093/pipeline/
- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by SeaRise** [comment]
> /run-sanitizer-test tsan

https://ci.pingcap.net/blue/organizations/jenkins/tiflash-sanitizer-daily/detail/tiflash-sanitizer-daily/1093/pipeline

File: dbms/src/Flash/executeQuery.cpp [hunk 1]
```diff
@@ -163,7 +163,7 @@ QueryExecutorPtr executeAsBlockIO(Context & context, bool internal)
 QueryExecutorPtr queryExecute(Context & context, bool internal)
 {
     if (context.getSettingsRef().enable_planner
-        && context.getSettingsRef().enable_pipeline
+        && (context.getSettingsRef().enable_pipeline || context.getSettingsRef().force_enable_pipeline)
```
Comment (windtalker) [inline dbms/src/Flash/executeQuery.cpp:166]: So `force_enable_pipeline` still does not take affect if `enable_planner == false`?

Comment (SeaRise) [reply dbms/src/Flash/executeQuery.cpp]: My mistake, fixed.
When `force_enable_pipeline` is true, the pipeline model will be used regardless of whether the planner is turned on.

File: dbms/src/Flash/Planner/Plans/PhysicalJoin.cpp [hunk 2]
```diff
@@ -121,6 +121,8 @@ PhysicalPlanNodePtr PhysicalJoin::build(
     tiflash_join.fillJoinOtherConditionsAction(context, left_input_header, right_input_header, probe_side_prepare_actions, original_probe_key_names, original_build_key_names, join_non_equal_conditions);
 
     const Settings & settings = context.getSettingsRef();
+    // Currently, the pipeline model does not support disk-based joins, so when force_enable_pipeline is true, the disk-based join will be disabled.
+    size_t max_bytes_before_external_join = settings.force_enable_pipeline ? 0 : static_cast<size_t>(settings.max_bytes_before_external_join);
```
Comment (windtalker) [inline dbms/src/Flash/Planner/Plans/PhysicalJoin.cpp:125]: Maybe add a log here?

Comment (SeaRise) [reply dbms/src/Flash/Planner/Plans/PhysicalJoin.cpp:125]: ok

Comment (SeaRise) [reply dbms/src/Flash/Planner/Plans/PhysicalJoin.cpp]: log added.
```
    size_t max_bytes_before_external_join = settings.max_bytes_before_external_join;
    if (settings.force_enable_pipeline && max_bytes_before_external_join > 0)
    {
        // Currently, the pipeline model does not support disk-based join, so when force_enable_pipeline is true, the disk-based join will be disabled.
        max_bytes_before_external_join = 0;
        LOG_WARNING(log, "Pipeline model does not support disk-based join, so set max_bytes_before_external_join = 0");
    }
```

**Review by gengliqi — APPROVED** [review]

**Pull request merged**